pkg/debug/common/common.go

```diff
@@ -0,0 +1,266 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/remotecommand"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/controllers/common"
+	"github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/pb"
+	"github.com/chaos-mesh/chaos-mesh/pkg/utils"
+	e2econfig "github.com/chaos-mesh/chaos-mesh/test/e2e/config"
+	"github.com/chaos-mesh/chaos-mesh/test/e2e/util/portforward"
+
+	kubectlscheme "k8s.io/kubectl/pkg/scheme"
+)
+
+var (
+	ColorReset = "\033[0m"
+	ColorRed   = "\033[31m"
+	ColorGreen = "\033[32m"
+	ColorCyan  = "\033[36m"
+	ColorBlue  = "\033[34m"
+	scheme     = runtime.NewScheme()
+)
+
+// ClientSet contains two different clients
+type ClientSet struct {
+	CtrlClient client.Client
+	K8sClient  *kubernetes.Clientset
+}
+
+func init() {
+	_ = v1alpha1.AddToScheme(scheme)
+	_ = clientgoscheme.AddToScheme(scheme)
+}
+
+func upperCaseChaos(str string) string {
+	parts := regexp.MustCompile("(.*)(chaos)").FindStringSubmatch(str)
+	return strings.Title(parts[1]) + strings.Title(parts[2])
+}
+
+// Print prints result to users in prettier format, with number of tabs and color specified
+func Print(s string, num int, color string) {
+	var tabStr string
+	for i := 0; i < num; i++ {
+		tabStr += "\t"
+	}
+	s = string(color) + s + string(ColorReset)
+	fmt.Printf("%s%s\n\n", tabStr, regexp.MustCompile("\n").ReplaceAllString(s, "\n"+tabStr))
+}
+
+// MarshalChaos returns json in readable format
+func MarshalChaos(s interface{}) (string, error) {
+	b, err := json.MarshalIndent(s, "", "  ")
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal indent: %s", err.Error())
+	}
+	return string(b), nil
+}
+
+// InitClientSet inits two different clients that would be used
+func InitClientSet() (*ClientSet, error) {
+	ctrlClient, err := client.New(config.GetConfigOrDie(), client.Options{Scheme: scheme})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create client")
+	}
+	k8sClient, err := kubernetes.NewForConfig(config.GetConfigOrDie())
+	if err != nil {
+		return nil, fmt.Errorf("error in getting access to K8S: %s", err.Error())
+	}
+	return &ClientSet{ctrlClient, k8sClient}, nil
+}
+
+// Exec executes certain command and returns the result
+// runtime-controller only support CRUD， use client-go client
+func Exec(pod string, ns string, cmd string, c *kubernetes.Clientset) (string, error) {
+	req := c.CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name(pod).
+		Namespace(ns).
+		SubResource("exec")
+
+	req.VersionedParams(&v1.PodExecOptions{
```

Comment:
`Exec` will fail with multi-container pod. We need specify a container name.

Reply:
thx for pointing it out!

I noticed only podChaos (the container is killed so nothing we couldn't exec I guess) and [timeChaos](https://chaos-mesh.org/docs/chaos_experiments/timechaos_experiment#configuration-file) would specify the container to be injected with `containerName`. So for the rest, is using the `Containers[0]` is enough for this?